dynatrace/src/main/java/com/dynatrace/opentelemetry/metric/Serializer.java

```diff
@@ -0,0 +1,188 @@
+/**
+ * Copyright 2021 Dynatrace LLC
+ *
+ * <p>Licensed under the Apache License, Version 2.0 (the "License"); you may not use this file
+ * except in compliance with the License. You may obtain a copy of the License at
+ *
+ * <p>http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * <p>Unless required by applicable law or agreed to in writing, software distributed under the
+ * License is distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either
+ * express or implied. See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package com.dynatrace.opentelemetry.metric;
+
+import com.dynatrace.metric.util.*;
+import com.google.common.annotations.VisibleForTesting;
+import io.opentelemetry.api.metrics.common.Labels;
+import io.opentelemetry.sdk.metrics.data.*;
+import java.time.Instant;
+import java.util.ArrayList;
+import java.util.List;
+import java.util.concurrent.TimeUnit;
+import java.util.logging.Logger;
+
+final class Serializer {
+  private static final Logger logger = Logger.getLogger(Serializer.class.getName());
+  // the precision used to identify whether a percentile is the 0% (min) or 100% (max) percentile.
+  private static final double PERCENTILE_PRECISION = 0.0001;
+  private static final String TEMPLATE_ERR_METRIC_LINE =
+      "Could not create metric line for data point with name %s (%s).";
+
+  private final MetricBuilderFactory builderFactory;
+
+  Serializer(MetricBuilderFactory builderFactory) {
+    this.builderFactory = builderFactory;
+  }
+
+  private Metric.Builder createMetricBuilder(MetricData metric, PointData point) {
+    return builderFactory
+        .newMetricBuilder(metric.getName())
+        .setDimensions(fromLabels(point.getLabels()))
+        .setTimestamp(Instant.ofEpochMilli(TimeUnit.NANOSECONDS.toMillis(point.getEpochNanos())));
```

Comment:
Could it be that only either of epochNanos and startEpochNanos is set? If so, we should check if one is 0 and fall back to the other.

Reply:
`epochNanos` is the time at which the point data was measured, while `startEpochNanos` is the time at which the instrument was created. Using `startTimeEpochNanos` might actually break collection (e. g. if the instrument is kept for a long time), so I would propose either having the `epochNanos` or no timestamp at all.

Reply:
I see, in this case using no timestamp in lieu of `epochNanos` makes sense since the ingest time is the better approximation here.